core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2434,8 +2434,15 @@ class SparkContext(config: SparkConf) extends Logging {
       val schedulingMode = getSchedulingMode.toString
       val addedJarPaths = addedJars.keys.toSeq
       val addedFilePaths = addedFiles.keys.toSeq
+      // SPARK-25392 pool Information should be stored in the event
+      val poolInformation = getAllPools.map { it =>
+        val xmlString = ("<pool><item PoolName=\"%s\" MinimumShare=\"%d\"" +
```

Comment:
Hmm, really don't like this kind of ad-hoc serialization format.

Instead, create a proper pool class for the REST API, and use that class to transfer pool information around.

Because this is a legacy event (not serialized by Jackson) you'll need some custom serialization in `JsonProtocol.scala`, but that's cleaner than what you have in this change.

Reply:
cc @vanzin Have changed the code accordingly, can you please have a look at this